### PR TITLE
gc: add scheduled gc task

### PR DESCRIPTION
**Pull request by Desiki-high**
- Add scheduled GC task, we will start GC every hour. 
- It will start the schedule after creating LocalAdapter. 
- The scheduled GC task will reduce local storage to less than 40 percent of the threshold.
Ref: #141

### CODE REVIEW

File: pkg/adapter/adapter.go [hunk 1]
```diff
@@ -65,6 +67,19 @@ func NewLocalAdapter(cfg *config.Config) (*LocalAdapter, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "create content provider")
 	}
+	c := cron.New()
+	threshold, err := humanize.ParseBytes(cfg.Provider.GCPolicy.Threshold)
+	if err != nil {
+		return nil, err
+	}
+	if _, err := c.AddFunc("@hourly", func() {
```
Comment (imeoer) [inline pkg/adapter/adapter.go:75]: Is there any reason to introduce a package to support cron jobs?

Comment (Desiki-high) [reply pkg/adapter/adapter.go:75]: > Is there any reason to use this package to support cron jobs?

Actually, just consider the convenience and possible new requirements in the future, have any better suggestions?

File: pkg/adapter/adapter.go [hunk 2]
```diff
@@ -65,6 +67,19 @@ func NewLocalAdapter(cfg *config.Config) (*LocalAdapter, error) {
...
+		return nil, err
+	}
+	// start scheduled gc task every hour
+	go func() {
+		for range ticker.C {
+			if err := content.GC(namespaces.WithNamespace(context.Background(), "acceleration-service"), (int64)(threshold/2)); err != nil {
+				logrus.Error(errors.Wrap(err, "scheduled gc task"))
+			}
+		}
+	}()
```
Comment (imeoer) [inline pkg/adapter/adapter.go:82]: Maybe we should put these codes into a separate func, then keep the NewLocalAdapter clearer.

File: pkg/content/content.go [hunk 3]
```diff
@@ -133,13 +133,17 @@ func (content *Content) Size() (int64, error) {
 }
 
 // GC clean the local caches by cfg.Provider.GCPolicy configuration
-func (content *Content) GC(ctx context.Context) error {
+func (content *Content) GC(ctx context.Context, threshold ...int64) error {
```
Comment (imeoer) [inline pkg/content/content.go:136]: Why treat the threshold as a slice here?

Comment (Desiki-high) [reply pkg/content/content.go:136]: ```scheduled gc task``` should clear more blobs, so ```GC``` should have different thresholds.

Comment (Desiki-high) [reply pkg/content/content.go:136]: We can change it to ```threshold int64```

File: pkg/content/content.go [hunk 4]
```diff
@@ -133,13 +133,13 @@ func (content *Content) Size() (int64, error) {
...
 // GC clean the local caches by cfg.Provider.GCPolicy configuration
-func (content *Content) GC(ctx context.Context) error {
+func (content *Content) GC(ctx context.Context, threshold int64) error {
 	size, err := content.Size()
 	if err != nil {
 		return err
 	}
 	// if the local content size over eighty percent of threshold, gc start
-	if size > (content.threshold*int64(80))/100 {
+	if size > (threshold*int64(80))/100 {
```
Comment (imeoer) [inline pkg/content/content.go:142]: let's change the `80` to a constant.

**Review by imeoer — APPROVED** [review]
LGTM!

**Pull request merged**